pkg/pkgrepository/app_deploy.go

```diff
@@ -30,13 +30,111 @@ func (a *App) deploy(tplOutput string) exec.CmdRunResult {
 			return exec.NewCmdRunResultWithErr(fmt.Errorf("Preparing kapp: %s", err))
 		}
 
-		return kapp.Deploy(tplOutput, a.startFlushingAllStatusUpdates, func(exec.CmdRunResult) {})
+		// TODO: should we inject this somewhere else? - there's some other rules that might interact with it...
+		fmt.Println("sending to kapp deploy: \n", appendRebaseRule(tplOutput))
+		return kapp.Deploy(appendRebaseRule(tplOutput), a.startFlushingAllStatusUpdates, func(exec.CmdRunResult) {})
 
 	default:
 		return exec.NewCmdRunResultWithErr(fmt.Errorf("Unsupported way to deploy"))
 	}
 }
 
+func appendRebaseRule(tplOutput string) string {
+	return tplOutput + `
+---
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+rebaseRules:
+- ytt:
+    overlayContractV1:
+      overlay.yml: |
+        #@ load("@ytt:data", "data")
+        #@ load("@ytt:yaml", "yaml")
+        #@ load("@ytt:json", "json")
+        #@ load("@ytt:overlay", "overlay")
+        #@ load("@ytt:struct", "struct")
+
+        #@ def get_rev(annotations):
+        #@  if hasattr(annotations, "packaging.carvel.dev/repo-pkg-revision"):
```

Comment:
annotation naming should not be Package specific, this also applies to PackageMetadata.

Reply:
(im still favoring `packaging.carvel.dev/revision` as a name)